src/index.js

```diff
@@ -1,15 +1,26 @@
 const arrify = require('arrify');
 const merge = require('deepmerge');
+const path = require('path');
 
 module.exports = ({ config }, options) => {
+  const NODE_MODULES = path.join(__dirname, 'node_modules');
+  const LOADER_EXTENSIONS = /\.vue$/;
+  const styleRule = config.module.rules.get('style');
+  const stylePostCssLoader = styleRule && styleRule.uses.get('postcss');
+  const lintRule = config.module.rules.get('lint');
+
   config.module.rule('vue')
-    .test(/\.vue$/)
+    .test(LOADER_EXTENSIONS)
     .use('vue')
-    .loader(require.resolve('vue-loader'))
-    .options(options);
+      .loader(require.resolve('vue-loader'))
+      .options(options);
+
+  config.resolve.extensions.add('.vue');
```

Comment:
This means we could say `import Component from 'component'` instead of `import Component from `component.vue`. I'm not sure this is a good idea, as I'd rather make it explicit that the file imported is a Vue component.